securing_openshift_gitops/managing-secrets-securely-using-sscsid-with-gitops.adoc

```diff
@@ -0,0 +1,41 @@
+:_mod-docs-content-type: ASSEMBLY
+include::_attributes/common-attributes.adoc[]
+[id="managing-secrets-securely-using-sscsid-with-gitops"]
+= Managing secrets securely using Secrets Store CSI driver with {gitops-shortname}
+:context: managing-secrets-securely-using-sscsid-with-gitops
+
+toc::[]
+
+This guide walks you through the process of integrating the Secrets Store Container Storage Interface (SSCSI) driver with the {gitops-shortname} Operator in {OCP} 4.14 and later.
+
+include::modules/gitops-managing-secrets-using-sscsid-with-gitops-overview.adoc[leveloffset=+1]
+
+[id="prerequisites_managing-secrets-securely-using-sscsid-with-gitops"]
+== Prerequisites
+* You have access to the cluster with `cluster-admin` privileges.
+* You have access to the {OCP} web console.
+* You have extracted and prepared the `ccoctl` binary.
+* You have installed the `jq` CLI tool.
+* Your cluster is installed on AWS and uses AWS Security Token Service (STS).
+* You have configured AWS Secrets Manager to store the required secrets.
+* link:https://docs.openshift.com/container-platform/latest/nodes/pods/nodes-pods-secrets-store.html#persistent-storage-csi-secrets-store-driver-install_nodes-pods-secrets-store[SSCSI Driver Operator is installed on your cluster].
+* {gitops-title} Operator is installed on your cluster.
+* You have a {gitops-shortname} repository ready to use the secrets.
+* You are logged in to the Argo CD instance by using the Argo CD admin account.
+
+include::modules/gitops-storing-aws-secret-manager-resources-in-gitops-repository.adoc[leveloffset=+1]
+
+include::modules/gitops-configuring-sscsi-driver-to-mount-secrets-from-aws-secrets-manager.adoc[leveloffset=+1]
+
+include::modules/gitops-configuring-gitops-managed-resources-to-use-mounted-secrets.adoc[leveloffset=+1]
+
+[id="additional-resources_managing-secrets-securely-using-sscsid-with-gitops"]
+[role="_additional-resources"]
+== Additional resources
+
+* link:https://github.com/redhat-developer/gitops-operator/blob/bfdf81944ea60b1d62d7de7c223c3bea3e7363a6/docs/Integrate%20GitOps%20with%20Secrets%20Management.md#obtain-the-ccoctl-tool[Obtaining the `ccoctl` tool]
+* link:https://docs.openshift.com/container-platform/latest/installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.html#cco-ccoctl-configuring_configuring-iam-ibm-cloud[Configuring the Cloud Credential Operator utility]
+* link:https://github.com/redhat-developer/gitops-operator/blob/bfdf81944ea60b1d62d7de7c223c3bea3e7363a6/docs/Integrate%20GitOps%20with%20Secrets%20Management.md#configure-your-aws-cluster-to-use-aws-security-token-service-sts[Configure your AWS cluster to use AWS STS]
```

Comment:
Afaik, migration of a non-STS cluster to use STS is not supported. I have raised this with CCO SMEs to get their input.

Reply:
Yes. In place migration of a cluster to STS is not supported and recommended. The github [readme](https://github.com/openshift/cloud-credential-operator/blob/master/docs/sts.md#steps-to-in-place-migrate-an-openshift-cluster-to-sts) says 
> NOTE This is just for developers interested in taking an existing cluster to STS. This is explicitly NOT RECOMMENED OR SUPPORTED.

Maybe we can update our doc with this information and recommend installing a OCP cluster with STS instead of doing in-place migration and point to the [whole doc](https://github.com/openshift/cloud-credential-operator/blob/master/docs/sts.md) instead of inplace-migration sub-section

Reply:
Sounds good

Reply:
Thanks @varshab1210 and @svghadi for your inputs.

We are not saying anywhere in the doc to migrate the non-STS cluster to STS. So no need to mention it I believe. WDYT?

We cannot give an external link to a GitHub site, in this case - https://github.com/openshift/cloud-credential-operator/blob/master/docs/sts.md, as this site is not owned or operated by RH. However, we can link to [this doc](https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-customizations.html#cco-ccoctl-configuring_installing-aws-customizations). The problem is that the steps are different and I need @ciiay and also someone from QE (cc: @varshab1210) to validate the steps in this document.

Reply:
Makes sense on not mentioning about migration. In that case, we just have to update the link once it's validated.

Reply:
Thanks folks - heard back from CCO SMEs confirming that migration is definitely not supported and the GitHub link to that unsupported procedure ought to be removed. 

The CCO GH repo mentioned [above](https://github.com/openshift/openshift-docs/pull/69291/files#r1440116819) is owned by RH afaik (all owners are RH devs, and CCO is an OCP component), but that's sort of beside the point. The parts of [this page](https://github.com/openshift/openshift-docs/pull/69291/files#r1440116819) that reflect conceptual information and officially supported procedures have already been covered in official product docs ([Conceptual](https://docs.openshift.com/container-platform/4.14/authentication/managing_cloud_provider_credentials/cco-short-term-creds.html#cco-short-term-creds-aws_cco-short-term-creds), [Procedural](https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-customizations.html#installing-aws-with-short-term-creds_installing-aws-customizations)).

Re: downstream docs, I see two potential approaches:

1. Link directly to [Configuring an AWS cluster to use short-term credentials](https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-customizations.html#installing-aws-with-short-term-creds_installing-aws-customizations) in the install docs. Ignore "long-term credentials" entirely; that is not related to STS at all. As you note, it's a long procedure... but that's how to configure a cluster to use STS at install. There's not a short way around it in OCP. **However**, given that already having a cluster with STS is a prerequisite to this task, I am not really sure that the install steps are helpful in this context.
2. It might be helpful for the user to be able to determine whether their cluster is using AWS STS, since it is required for the task. There is information about that here: [Determining the Cloud Credential Operator mode](https://docs.openshift.com/container-platform/4.14/authentication/managing_cloud_provider_credentials/about-cloud-credential-operator.html#cco-determine-mode_about-cloud-credential-operator)

Personally, I would lean towards approach 2. This task is for an installed cluster (IIUC), so pointing them to install docs seems unrelated.

Reply:
Hi @Srivaralakshmi , I agree with @jeana-redhat . We can provide a pointer to help users to determine if their cluster is using AWS STS instead of educating them how to set up a cluster using STS as it's out of the scope of this documentation.

Reply:
Thanks @jeana-redhat for the clarification and suggestions. Thanks @varshab1210 and @ciiay for the confirmation.

I will raise a JIRA for this and update the link in the new PR and reference this issue and PR number in it.